### PR TITLE
use exported error redis.Nil

### PR DESCRIPTION
**Pull request by kavehjamshidi**

### CODE REVIEW

File: slidingwindow.go [hunk 1]
```diff
@@ -136,7 +136,7 @@ func (s *SlidingWindowRedis) Increment(ctx context.Context, prev, curr time.Time
 	case <-done:
 		if err == redis.TxFailedErr {
 			return 0, 0, errors.Wrap(err, "redis transaction failed")
-		} else if err != nil && err.Error() == "redis: nil" { // TODO: is there an exported error of that type?
+		} else if err != nil && err == redis.Nil {
```
Comment (mennanov) [inline slidingwindow.go:139]: I know this was there before but is the check for `err != nill` necessary here? Can we just do this?

```golang
} else if err == redis.Nil {
```

**Review by mennanov — CHANGES_REQUESTED** [review]
Good catch! 
Do you think it is possible to add a test for this case?

File: slidingwindow.go (hunk 1 above)
Comment (kavehjamshidi) [reply slidingwindow.go]: I fixed this, but I have to check the tests to see if I can do what you asked regarding a test case to cover this branch

Comment (mennanov) [reply slidingwindow.go]: Ok, let me know if you are willing to add such a test, otherwise i will merge as is and add a test later

Comment (kavehjamshidi) [reply slidingwindow.go]: merge this as it is, please

**Review by mennanov — APPROVED** [review]

**Comment by mennanov** [comment]
Thanks for contributing!

**Pull request merged**